GiniVision/Classes/Core/GiniVisionDocument.swift

```diff
@@ -126,11 +118,10 @@ extension GiniVisionDocument {
      - Throws: `DocumentValidationError.exceededMaxFileSize` is thrown if the document is not valid.
      
      */
-    public func validate() throws {
-        let document = self
+    public class func validate(_ document: GiniVisionDocument, withConfig giniConfiguration: GiniConfiguration) throws {
```

Comment:
The old signature should be deprecated and add a new method with new signature.

Reply:
True. I will revert it back and add the deprecation attribute.